rfc/rfc-8/rfc-8.md

```diff
@@ -0,0 +1,209 @@
+be extended and shared for this partition also. The type field will be used to detect the record_index payload record.
+Here is the schema for the record_index payload record.
+```
+    {
+        "name": "recordIndexMetadata",
+        "doc": "Metadata Index that contains information about record keys and their location in the dataset",
+        "type": [
+            "null",
+             {
+               "type": "record",
+               "name": "HoodieRecordIndexInfo",
+                "fields": [
+                    {
+                        "name": "partition",
+                        "type": "string",
+                        "doc": "Partition which contains the record",
+                        "avro.java.string": "String"
+                    },
+                    {
+                        "name": "fileIdHighBits",
+                        "type": "long",
+                        "doc": "fileId which contains the record (high 64 bits)"
+                    },
+                    {
+                        "name": "fileIdLowBits",
+                        "type": "long",
+                        "doc": "fileId which contains the record (low 64 bits)"
+                    },
+                    {
+                        "name": "fileIndex",
+                        "type": "int",
+                        "doc": "index of the file"
+                    },
+                    {
+                        "name": "instantTime",
+                        "type": "long",
+                        "doc": "Epoch time in millisecond at which record was added"
+                    }
+                ]
+            }
+        ],
+        "default" : null
+    }
+```
+
+The key for the record index record would be the actual key from the record. The partition name is also saved as string.
+HUDI base files names have a format which includes a UUID fileID, an integer file Index, a write token and a timestamp. 
+The record index payload only saves the fileID and file index information. The fileID is split into UUID and the integer file index. The UUID is encoded into two longs and the file index is saved
+as an integer. The timestamp is encoded into epoch time in milliseconds.
+
+This schema format is chosen to minimize the data size of each mapping to ensure the smallest possible size of the 
+record index even for datasets with billions of records. 
+
+Experiments have shown that with random UUID record keys and datestr partitions (YYYY/MM/DD), we can achieve an average
+size of 50 to 55 bytes per mapping saved in the record index. The size might even be lower for keys which may compress better.
+
+<img src="metadata_index_bloom_partition.png" alt="Bloom filter partition" width="500"/>
+
+Below picture gives a pictorial representation of record index partition in metadata table.
+<img src="metadata_index_col_stats.png" alt="Column Stats Partition" width="480"/>
```

Comment:
The image here shows the column stats partition, not the record index.

Reply:
Fixed